x/auth/client/rest/query.go

```diff
@@ -204,3 +194,21 @@ func packStdTxResponse(w http.ResponseWriter, clientCtx client.Context, txRes *s
 
 	return nil
 }
+
+func checkForJSONMarshalFailure(w http.ResponseWriter, ctx client.Context, resp interface{}) error {
+	// LegacyAmino used intentionally here to error message
+	const errMsg = "unregistered concrete type"
+	marshaler := ctx.LegacyAmino
+
+	_, err := marshaler.MarshalJSON(resp)
+
+	if err != nil && strings.Contains(err.Error(), errMsg) {
+		rest.WriteErrorResponse(w, http.StatusInternalServerError,
+			"This transaction was created with the new SIGN_MODE_DIRECT signing method, and therefore cannot be displayed"+
```

Comment:
it would be great to keep "gRPC gateway endpoint is ..." at the end of the error message.